redis/src/test/scala/zio/redis/ApiSpec.scala

```diff
@@ -39,10 +44,204 @@ object ApiSpec extends BaseSpec {
+            value           <- uuid
+            _               <- set(key, value, None, None, None)
+            scan            <- scan(0L, None, None, None)
+            (next, elements) = scan
+          } yield assert(next)(isNonEmptyString) && assert(elements)(isNonEmpty)
+        },
+        testM("fetch random key") {
+          for {
+            key       <- uuid
+            value     <- uuid
+            _         <- set(key, value, None, None, None)
+            allKeys   <- keys("*")
+            randomKey <- randomKey()
+          } yield assert(allKeys)(contains(randomKey.get))
+        } @@ ignore,
+        testM("dump followed by restore") {
+          for {
+            key      <- uuid
+            value    <- uuid
+            _        <- set(key, value, None, None, None)
+            dumped   <- dump(key)
+            _        <- del(key, Nil)
+            restore  <- restore(key, 0L, dumped, None, None, None, None).either
+            restored <- get(key)
+          } yield assert(restore)(isRight) && assert(restored)(isSome(equalTo(value)))
+        } @@ ignore,
+        suite("expiration")(
+          testM("pExpire followed by pTtl") {
+            for {
+              key   <- uuid
+              value <- uuid
+              _     <- set(key, value, None, None, None)
+              exp   <- pExpire(key, 100.millis)
+              ttl   <- pTtl(key)
+            } yield assert(exp)(isTrue) && assert(ttl.toMillis)(isPositive)
+          },
+          testM("pExpireAt followed by exists and pTtl") {
+            for {
+              key       <- uuid
+              value     <- uuid
+              expiresAt <- instantOf(1000)
+              _         <- set(key, value, None, None, None)
+              _         <- pExpireAt(key, expiresAt)
+              response1 <- exists(key, Nil)
+              ttl       <- pTtl(key)
+              _         <- ZIO.sleep(1010.millis)
+              response2 <- exists(key, Nil)
+            } yield assert(response1)(isTrue) && assert(response2)(isFalse) && assert(ttl.toMillis)(isPositive)
+          },
+          testM("expire followed by ttl and persist") {
+            for {
+              key       <- uuid
+              value     <- uuid
+              _         <- set(key, value, None, None, None)
+              _         <- expire(key, 10000.millis)
+              ttl1      <- ttl(key)
+              persisted <- persist(key)
+              ttl2      <- ttl(key).either
+            } yield assert(ttl1.toMillis)(isPositive) && assert(persisted)(isTrue) && assert(ttl2)(isLeft)
+          },
+          testM("expireAt followed by get and ttl") {
+            for {
+              key       <- uuid
+              value     <- uuid
+              expiresAt <- instantOf(2000)
+              _         <- set(key, value, None, None, None)
+              _         <- expireAt(key, expiresAt)
+              response1 <- exists(key, Nil)
+              ttl       <- ttl(key)
```

Comment:
I don't like this test, as it mixes concerns. Checking the ttl should be done separately. If you can do it quickly, go for it, otherwise, let's do it as follow up.